ci/deploy/linux.py

```diff
@@ -38,12 +38,14 @@ def main() -> None:
         publish_deb("materialized-unstable", deb.unstable_version(workspace))
 
     print(f"--- Tagging Docker images")
+    deps = repo.resolve_dependencies(image for image in repo if image.publish)
+    deps.acquire()
     if buildkite_tag:
-        tag_docker(repo, buildkite_tag)
-        tag_docker_latest_maybe(repo, buildkite_tag)
+        deps.push_tagged(buildkite_tag)
+        tag_docker_latest_maybe(deps, buildkite_tag)
     else:
-        tag_docker(repo, f'unstable-{git.rev_parse("HEAD")}')
-        tag_docker(repo, "unstable")
+        deps.push_tagged(f'unstable-{git.rev_parse("HEAD")}')
+        deps.push_tagged("unstable")
```

Comment:
This `else` bit happens in build.py now so it shouldn't happen here, by my read.